Update Install-ChocoBinaries.ps1

kendr1ck opened the pull request:
Fixes SSL/TLS error when trying to download install.ps1 from https://chocolatey.org. See powershell command listed at https://chocolatey.org/docs/installation#install-from-powershell-v3.

kendr1ck committed:
Update Install-ChocoBinaries.ps1

Fixes SSL/TLS error when trying to download install.ps1 from https://chocolatey.org. See powershell command listed at https://chocolatey.org/docs/installation#install-from-powershell-v3.

ethanbergstrom commented:
Hey @kendr1ck from the link you provided I'm guessing you're seeing this issue on PowerShell 2.0 or below due to the TLS 1.2 issue? 

The module manifest inherited a [Powershell 3.0 minimum requirement](https://github.com/ethanbergstrom/Chocolatier/blob/master/Chocolatier.psd1#L8) from ChocolateyGet, so was wondering what situations you were having the issue with today, and if the module manifest would need to be adjusted accordingly.

kendr1ck commented:
I actually saw this on a newly launched AWS Server 2016 instance using
their default machine image. I was successfully using ChocolateyGet with
those images as of a couple weeks ago. I switched to your module when I
went to make a pull request on ChocoatelyGet and saw your pull request from
November. Adding that line to both modules seems to fix it.

-- 

Mike Carroll | Lead Engineer | Vesper Data and Technology Services
d: +19083428357 | c: +19086056271 | e: mcarroll@vesperdata.com

On Fri, Feb 7, 2020, 7:13 PM Ethan Bergstrom <notifications@github.com>
wrote:

> Hey @kendr1ck <https://github.com/kendr1ck> from the link you provided
> I'm guessing you're seeing this issue on PowerShell 2.0 due to the TLS 1.2
> issue?
>
> The module manifest inherited a Powershell 3.0 minimum requirement from
> ChocolateyGet, so was wondering what situations you were having the issue
> with today, and if the module manifest would need to be adjusted
> accordingly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethanbergstrom/Chocolatier/pull/3?email_source=notifications&email_token=AIXBZ6EG3QXFJCJKRFYJZ63RBX2L5A5CNFSM4KRSZWN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELFBR6Q#issuecomment-583670010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIXBZ6BBQDUAA4B7JIPV4F3RBX2L5ANCNFSM4KRSZWNQ>
> .
>

ethanbergstrom commented:
Huh...weird. I'll try to take a closer look this weekend.

ethanbergstrom committed:
Use semantic bit names, and not override defaults

Older versions of PowerShell / .NET are opinionated about which ciphers to support, while newer versions (ex: Build 19559) default to whatever ciphers the OS supports.

If .NET isn't falling back on the OS defaults, explicitly add TLS 1.2 as a supported cipher for this session, otherwise let the OS take care of it.

ethanbergstrom commented:
Merged in the changes and uploaded the new version to PSGallery. Thanks for bringing this issue to my attention!

Tested on a pure Server 2016 EC2 instance:
![image](https://user-images.githubusercontent.com/13603162/74089980-1fe13e00-4a6c-11ea-9e22-a077bb932093.png)

kendr1ck commented:
Thanks for the quick fix! And for continuing the work started by the
ChocolateyGet module. I'm just starting to get into using DSC heavily and
have come to really depend on that module.

On Sat, Feb 8, 2020, 1:02 PM Ethan Bergstrom <notifications@github.com>
wrote:

> Merged in the changes and uploaded the new version to PSGallery. Thanks
> for bringing this issue to my attention!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethanbergstrom/Chocolatier/pull/3?email_source=notifications&email_token=AIXBZ6COZZNB4BIYH2PUUDTRB3XUBA5CNFSM4KRSZWN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELFX35I#issuecomment-583761397>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIXBZ6HI2WVIA24SHLXJJS3RB3XUBANCNFSM4KRSZWNQ>
> .
>
